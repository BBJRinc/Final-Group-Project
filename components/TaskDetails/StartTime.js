import React, { Component } from 'react'
import {
    DatePickerIOS, Platform,
    View,
    StyleSheet,
    Text,
    Picker, StatusBar, Modal, TouchableHighlight, Item
} from 'react-native'
import IconI from 'react-native-vector-icons/Ionicons';
import moment from 'moment'

export default class StartTime extends Component {
    constructor(props) {
        super(props);
        this.state = { chosenDate: new Date() };

        this.setDate = this.setDate.bind(this);
    }

    setDate(newDate) {
        this.setState({ chosenDate: newDate });
        this.props.setTaskStartTime(newDate)
    }

    render() {
        return (
            <View style={styles.container}>
                <Modal
                    animationType="slide"
                    transparent={false}
                    onRequestClose={() => this.props.setStartTimePicker()}
                    visible={this.props.showStartTimePicker}
                >
                    <DatePickerIOS
                        date={this.state.chosenDate}
                        onDateChange={this.setDate}
                        placeHolder='Select start time'
                        mode='time'
                        minuteInterval={15}
                    />
                    <TouchableHighlight style={{ alignItems: 'center' }}
                        onPress={() => {
                            this.props.setStartTimePicker();
                            }}>
                            <Text style={{height:30}}>hello</Text>
                    </TouchableHighlight>
                </Modal>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        ...Platform.select({
          ios: {
            // backgroundColor: 'Black',
          },
          android: {
            // backgroundColor: 'Black',
            color:'black'
          }
        })
    }
})
