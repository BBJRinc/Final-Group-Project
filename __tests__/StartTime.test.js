//Brandons Test

import React from 'react';
import StartTime from '../components/TaskDetails/StartTime'

import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<StartTime />).toJSON();
    expect(tree).toMatchSnapshot();
});