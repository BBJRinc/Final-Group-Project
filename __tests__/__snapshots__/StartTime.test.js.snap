// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={true}
  >
    <View
      style={undefined}
    >
      <RCTDatePicker
        date={1522974473473}
        locale={undefined}
        maximumDate={undefined}
        minimumDate={undefined}
        minuteInterval={15}
        mode="time"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 216,
          }
        }
        timeZoneOffsetInMinutes={undefined}
      />
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "height": 30,
          }
        }
      >
        hello
      </Text>
    </View>
  </Modal>
</View>
`;
